src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -761,12 +771,34 @@ void CCalcEngine::ResolveHighestPrecedenceOperation()
         {
             m_currentVal = m_holdVal;
             DisplayNum(); // to update the m_numberString
-            m_HistoryCollector.AddBinOpToHistory(m_nOpCode, false);
-            m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal); // Adding the repeated last op to history
+            if (IsBinOpCode(m_nOpCode))
+            {
+                m_HistoryCollector.AddBinOpToHistory(m_nOpCode, false);
+                m_HistoryCollector.AddOpndToHistory(m_numberString, m_currentVal); // Adding the repeated last op to history
+            }
+            else if (m_nOpCode == IDC_SIGN || IsUnaryOpCode(m_nOpCode))
```

Comment:
The code here (and below) will have to change as a result of the feedback above.